src/wasm/wasm-validator.cpp

```diff
@@ -2026,20 +2026,35 @@ void FunctionValidator::visitTry(Try* curr) {
       curr->type,
       curr->body,
       "try's type does not match try body's type");
-    shouldBeSubTypeOrFirstIsUnreachable(
-      curr->catchBody->type,
-      curr->type,
-      curr->catchBody,
-      "try's type does not match catch's body type");
+    for (auto catchBody : curr->catchBodies) {
+      shouldBeSubTypeOrFirstIsUnreachable(
+        catchBody->type,
+        curr->type,
+        catchBody,
+        "try's type does not match catch's body type");
+    }
   } else {
     shouldBeEqual(curr->body->type,
                   Type(Type::unreachable),
                   curr,
                   "unreachable try-catch must have unreachable try body");
-    shouldBeEqual(curr->catchBody->type,
-                  Type(Type::unreachable),
+    for (auto catchBody : curr->catchBodies) {
+      shouldBeEqual(catchBody->type,
+                    Type(Type::unreachable),
+                    curr,
+                    "unreachable try-catch must have unreachable catch body");
+    }
+  }
+  if (curr->hasCatchAll()) {
```

Comment:
`hasCatchAll` only returns true if the difference is 1, so this check will never fail. Perhaps it could be replaced with an assertion?

Reply:
Oh, good catch. In the first version I maintained `hasCatchAll` a separate boolean variable within `Try` and later removed it because it basically contained redundant info, and then I mechanically replaced `hasCatchAll` with `hasCatchAll()`, which resulted in this.. I changed the code to this, WDYT?
```cpp
  shouldBeTrue(curr->catchBodies.size() - curr->catchEvents.size() <= 1,          
               curr,                                                              
               "the number of catch blocks and events do not match");
```

Reply:
Looks good!